packages/frontend/src/components/EditorSettings/Notifications.tsx

```diff
@@ -0,0 +1,146 @@
+import { IFlow } from '@plumber/types'
+
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { useMutation, useQuery } from '@apollo/client'
+import { Flex, Skeleton, Stack, Text } from '@chakra-ui/react'
+import { Menu, useToast } from '@opengovsg/design-system-react'
+import { UPDATE_FLOW_CONFIG } from 'graphql/mutations/update-flow-config'
+import { GET_FLOW } from 'graphql/queries/get-flow'
+
+enum Frequency {
+  Once = 'once',
+  Always = 'always',
+}
+
+const frequencyOptions = [
+  {
+    label: 'First error on this pipe, one email notification per day',
+    value: Frequency.Once,
+  },
+  {
+    label: 'Each error on this pipe',
+    value: Frequency.Always,
+  },
+]
+
+const defaultFrequency = Frequency.Once
+
+export default function Notifications() {
+  const { flowId } = useParams()
+  const [frequency, setFrequency] = useState(defaultFrequency)
+  const { data, loading } = useQuery(GET_FLOW, { variables: { id: flowId } })
+  const flow: IFlow = data?.getFlow
+  const [updateFlowConfig] = useMutation(UPDATE_FLOW_CONFIG)
+  const toast = useToast()
+
+  const onFlowConfigUpdate = useCallback(
+    async (frequency: Frequency) => {
+      await updateFlowConfig({
+        variables: {
+          input: {
+            id: flowId,
+            notificationFrequency: frequency,
+          },
+        },
+        optimisticResponse: {
+          updateFlowConfig: {
+            __typename: 'Flow',
+            id: flow?.id,
+            config: {
+              errorConfig: {
+                notificationFrequency: frequency,
+              },
+            },
+          },
+        },
+      })
+
+      const displayLabel = frequencyOptions
+        .find((option) => option.value === frequency)
+        ?.label.toLowerCase()
+
+      toast({
+        title: `You will now receive an email notification for ${displayLabel}`,
+        status: 'success',
+        duration: 3000,
+        isClosable: true,
+        position: 'top',
+      })
+    },
+    [flow?.id, flowId, updateFlowConfig, toast],
+  )
+
+  const handleClick = useCallback(
+    (frequencyOption: Frequency) => {
+      // only update flow config if a different option is selected
+      if (frequencyOption !== frequency) {
+        setFrequency(frequencyOption)
+        onFlowConfigUpdate(frequencyOption)
+      }
+    },
+    [onFlowConfigUpdate, frequency],
+  )
+
+  useEffect(() => {
+    // Retrieve frequency setting if config was set previously, else set default frequency in config
+    if (!flow) {
+      return
+    }
+    if (flow.config?.errorConfig?.notificationFrequency) {
+      setFrequency(flow.config.errorConfig.notificationFrequency as Frequency)
+    } else {
+      onFlowConfigUpdate(defaultFrequency)
```

Comment:
Is there a need to forcibly update the flow when the user accesses this page? I want to try to get rid of `useEffect` if possible. It also saves a mutation and a re-render.

In the backend, if `notificationFrequency` is undefined, we default to`once-per-day`. We can use the same philosophy in the frontend and default to `once-per-day` if `flow.config?.errorConfig?.notificationFrequency` is undefined.

Reply:
ok lemme try to use the db as the source of truth instead, thanks!

Reply:
ohya can i argue that the reason I forcibly update the flow is because my idea is that if users are accessing this page, we should definitely record it down in the config incrementally instead of just manually adding it for every flow. I believe the mutation is quite minimal in terms of time taken cus its just a JSON field? @ogp-weeloong

Reply:
ok i fixed this alr!